maven-core/src/main/java/org/apache/maven/settings/SettingsUtils.java

```diff
@@ -51,9 +37,9 @@ private SettingsUtils()
      * @param recessive
      * @param recessiveSourceLevel
      */
-    public static Settings merge( Settings dominant, Settings recessive, String recessiveSourceLevel )
+    public static void merge( Settings dominant, Settings recessive, String recessiveSourceLevel )
     {
-        return new MavenSettingsMerger().merge( dominant, recessive, recessiveSourceLevel );
+        SettingsUtilsV4.merge( dominant.getDelegate(), recessive.getDelegate(), recessiveSourceLevel );
```

Comment:
`dominant` and `recessive` can be null in v3.

caller expect that `dominant` will have a result of merge , so we need

```
dominant.delegate = SettingsUtilsV4.merge( ...
```

Reply:
The method now reads:
```
     public static void merge(Settings dominant, Settings recessive, String recessiveSourceLevel) {
         if (dominant != null && recessive != null) {
             dominant.delegate = SettingsUtilsV4.merge(dominant.getDelegate(), recessive.getDelegate());
         }
     }
```

As for the source level, I think it would be better to leverage the InputLocationTracker in the model, which is way more powerful, for both settings and toolchains.  And get rid of the TrackableBase in the v4 models.